outputs/openapi/components.go

```diff
@@ -11,3 +15,37 @@ func makeComponentRef(name, pkg string) string {
 func makeComponentRefName(name, pkg string) string {
 	return strings.ReplaceAll(pkg, "/", "_") + "." + name
 }
+
+func mapComponentEnums(component astra.Field) []interface{} {
+	var enums []interface{}
+	for _, enum := range component.EnumValues {
+		schema := mapAcceptedType(component.Type)
+		switch schema.Type {
```

Comment:
I think this conversion back to the original type should be done in `astTraversal/type.go` and `component.EnumValues` should be a list of generics, this is because when other output types are added we will probably have to repeat this code breaking the DRY principles. I also feel it probably makes more sense there and we can use `token.<type>` instead of our string based case statement and allows the output implementation to handle how they are represented if we give them the underlying value in their original type?

Reply:
.type is impossible to use, as it always parses as a string so we absolutely need to do the method we do at the moment. But I see your point for moving it to type.go

Reply:
Actually managed to figure it out, which worked out very well. Turns out we can switch the .Kind() which we can strongly type

Reply:
That's really cool, do you think we can use this more in the rest of the code base? Might make things more obvious/ less error prone?

Reply:
We can scope to see if it should be used anywhere